### PR TITLE
Using padding in DATA frame for budget calculations

### PR DESCRIPTION
**Pull request by jitsni**

### CODE REVIEW

File: src/main/java/org/reaktivity/nukleus/http/internal/stream/ClientAcceptStream.java [hunk 1]
```diff
@@ -309,15 +309,15 @@ private void processData(
     {
         DataFW data = factory.dataRO.wrap(buffer, index, index + length);
 
-        sourceWindow -= data.length();
+        sourceWindow -= data.length() + data.padding();
```
Comment (jfallows) [inline src/main/java/org/reaktivity/nukleus/http/internal/stream/ClientAcceptStream.java]: `sourceWindow` -> `sourceBudget`

File: src/main/java/org/reaktivity/nukleus/http/internal/stream/ClientConnectReplyStream.java [hunk 2]
```diff
@@ -698,7 +680,7 @@ private int decodeHttpData(
 
         if (writableBytes > 0)
         {
-            factory.writer.doHttpData(acceptReply, acceptReplyId, payload, offset, writableBytes);
+            factory.writer.doHttpData(acceptReply, acceptReplyId, acceptReplyWindowPadding, payload, offset, writableBytes);
```
Comment (jfallows) [inline src/main/java/org/reaktivity/nukleus/http/internal/stream/ClientConnectReplyStream.java]: `acceptReplyWindowPadding` -> `acceptReplyPadding`

File: src/main/java/org/reaktivity/nukleus/http/internal/stream/ClientConnectReplyStream.java [hunk 3]
```diff
@@ -698,7 +680,7 @@ private int decodeHttpData(
...
             acceptReplyWindowBudget -= writableBytes + acceptReplyWindowPadding;
```
Comment (jfallows) [inline src/main/java/org/reaktivity/nukleus/http/internal/stream/ClientConnectReplyStream.java]: `acceptReplyWindowBudget` -> `acceptReplyBudget`

File: src/main/java/org/reaktivity/nukleus/http/internal/stream/ClientConnectReplyStream.java [hunk 4]
```diff
@@ -953,28 +934,12 @@ private void handleWindow(
...
-        }
-        doSendWindowIfAligned();
-    }
-
-    private void doSendWindowIfAligned()
-    {
-        // Don't send WINDOW( ,connectReplyWindowPadding) until the budgets are aligned
-        // Otherwise padding may be applied at different times on both sides
-        if (connectReplyWindowAligned)
+        final int connectReplyWindowCredit = acceptReplyWindowBudget - connectReplyWindowBudget;
```
Comment (jfallows) [inline src/main/java/org/reaktivity/nukleus/http/internal/stream/ClientConnectReplyStream.java]: `connectReplyWindowCredit` -> `connectReplyCredit`
`connectReplyWindowBudget` -> `connectReplyBudget`

File: src/main/java/org/reaktivity/nukleus/http/internal/stream/ServerAcceptStream.java [hunk 5]
```diff
@@ -796,21 +797,21 @@ private int decodeHttpData(
 
         // TODO: consider chunks
         int writableBytes = Math.min(length, contentRemaining);
-        writableBytes = Math.min(targetWindowBudget, writableBytes);
+        writableBytes = Math.min(targetWindowBudget - targetWindowPadding, writableBytes);
```
Comment (jfallows) [inline src/main/java/org/reaktivity/nukleus/http/internal/stream/ServerAcceptStream.java]: `targetWindowBudget` -> `targetBudget`
`targetWindowPadding` -> `targetPadding`

File: src/main/java/org/reaktivity/nukleus/http/internal/stream/ServerConnectReplyStream.java [hunk 6]
```diff
@@ -58,7 +58,6 @@
     private boolean endDeferred;
 
     private int connectReplyWindowBudget;
```
Comment (jfallows) [inline src/main/java/org/reaktivity/nukleus/http/internal/stream/ServerConnectReplyStream.java]: `connectReplyWindowBudget` -> `connectReplyBudget`

File: src/main/java/org/reaktivity/nukleus/http/internal/stream/ServerConnectReplyStream.java [hunk 7]
```diff
@@ -167,7 +166,7 @@ private void streamAfterRejectOrReset(
             DataFW data = factory.dataRO.wrap(buffer, index, index + length);
             final long streamId = data.streamId();
             connectReplyWindowBudget += data.length();
-            factory.writer.doWindow(connectReplyThrottle, streamId, data.length(), connectReplyWindowPadding);
+            factory.writer.doWindow(connectReplyThrottle, streamId, data.length() + data.padding(), 0);
```
Comment (jfallows) [inline src/main/java/org/reaktivity/nukleus/http/internal/stream/ServerConnectReplyStream.java]: Deliberate switch to `0` padding here?

File: src/main/java/org/reaktivity/nukleus/http/internal/stream/ServerConnectReplyStream.java [hunk 8]
```diff
@@ -246,7 +245,7 @@ private void processBegin(
                             RESPONSE_HEADERS_TOO_LONG_RESPONSE.length + acceptState.acceptReplyWindowPadding;
                     assert acceptState.acceptReplyWindowBudget >= 0;
                     factory.writer.doData(acceptState.acceptReply, acceptState.replyStreamId,
-                                  slot, 0, RESPONSE_HEADERS_TOO_LONG_RESPONSE.length);
+                            acceptState.acceptReplyWindowPadding, slot, 0, RESPONSE_HEADERS_TOO_LONG_RESPONSE.length);
```
Comment (jfallows) [inline src/main/java/org/reaktivity/nukleus/http/internal/stream/ServerConnectReplyStream.java]: `acceptReplyWindowBudget` -> `acceptReplyBudget`
`acceptReplyWindowPadding` -> `acceptReplyPadding`

File: src/main/java/org/reaktivity/nukleus/http/internal/stream/ServerConnectReplyStream.java (hunk 7 above)
Comment (jitsni) [reply src/main/java/org/reaktivity/nukleus/http/internal/stream/ServerConnectReplyStream.java]: Yes, since the data is just consumed.

**Pull request merged**